Challenge-4/Bot_1/src/helper.ts

```diff
@@ -0,0 +1,88 @@
+import { ethers } from "forta-agent";
+import { LRUCache } from "lru-cache";
+import {
+  CONFIGURATOR_PROXY,
+  CONFIGURATOR_PROXY_ARB,
+  ERC20_ABI,
+} from "./constants";
+
+export default class Helper {
+  private provider: ethers.providers.Provider;
+  private collateralAssetsCache: LRUCache<string, { [name: string]: string }>;
+  private collateralNamesCache: LRUCache<string, string>;
+
+  constructor(provider: ethers.providers.Provider) {
+    this.provider = provider;
+    this.collateralAssetsCache = new LRUCache<
+      string,
+      { [name: string]: string }
+    >({
+      max: 1000,
+    });
+    this.collateralNamesCache = new LRUCache<string, string>({
+      max: 1000,
+    });
+  }
+
+  public async getCollateralAssets(
+    usdcAddress: string,
+    usdcAbi: string[],
+    blockNumber: number,
+  ): Promise<{ [name: string]: string }> {
+    const assetContract = new ethers.Contract(
+      usdcAddress,
+      usdcAbi,
+      this.provider,
+    );
+    const key = `${usdcAddress}`;
```

Comment:
If the `collateralAssets` have the possibility of changing, then the `key` would need the `blockNumber` like we usually include because it could change from one block to another.

Pardon if the comment from before was not clear, I just meant the fetching of names didn't need the block number. This `key` would still need it (possibly).